src/utils/viewer/viewer.cpp

```diff
@@ -382,42 +391,46 @@ Viewer::Viewer(const Arguments& arguments)
   printHelpText();
 }  // end Viewer::Viewer
 
-void Viewer::addObject(int ID) {
+int Viewer::addObject(int ID) {
   const std::string& configHandle =
       simulator_->getObjectAttributesManager()->getObjectHandleByID(ID);
-  addObject(configHandle);
+  return addObject(configHandle);
 }  // addObject
 
-void Viewer::addObject(const std::string& configFile) {
+int Viewer::addObject(const std::string& objectAttrHandle) {
   // Relative to agent bodynode
   Mn::Matrix4 T = agentBodyNode_->MagnumObject::transformationMatrix();
   Mn::Vector3 new_pos = T.transformPoint({0.1f, 1.5f, -2.0f});
 
-  int physObjectID = simulator_->addObjectByHandle(configFile);
+  int physObjectID = simulator_->addObjectByHandle(objectAttrHandle);
   simulator_->setTranslation(new_pos, physObjectID);
-  simulator_->setRotation(esp::core::randomRotation(), physObjectID);
+  simulator_->setRotation(Mn::Quaternion::fromMatrix(T.rotationNormalized()),
```

Comment:
> set to agent orientation

I would like to learn a bit more about this design choice. Why not set it to align the stage orientation?

Reply:
We could do that also. Just aiming for something consistent here. 

My reasoning: We are setting position from agent transform, so do the same for orientation. This way you always know that the object's local forward is aligned with the camera forward when it is added. 

Another choice: we could apply no transformation to added objects by default and provide another function to set an object transform (with orientation optional) relative to the agent as a follow-up operation. Do you have a preference?